codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlShapeSerVisitor.java

```diff
@@ -183,17 +184,18 @@ protected void serializeUnion(GenerationContext context, UnionShape shape) {
             Set<MemberShape> members = new TreeSet<>(shape.getAllMembers().values());
             for (MemberShape member : members) {
                 Shape target = context.getModel().expectShape(member.getTarget());
-                Symbol memberSymbol = symbolProvider.toSymbol(member);
-                String exportedMemberName = symbol.getName() + symbolProvider.toMemberName(member);
+                Symbol memberSymbol = SymbolUtils.createValueSymbolBuilder(
+                        symbolProvider.toMemberName(member),
+                        symbol.getNamespace()
+                ).build();
 
-                writer.openBlock("case *$L:", "", exportedMemberName, () -> {
+                writer.openBlock("case *$T:", "", memberSymbol, () -> {
                     writer.write("av := value.MemberElement($S)", memberSymbol.getName());
                     target.accept(getMemberSerVisitor(member, "uv.Value()", "av"));
```

Comment:
Another fix is required for `MemberElement` usage on line 190. Instead of passing in the string, have to construct a xml start element and pass it in.